File: src/main/java/gg/climb/lolobjects/esports/League.java
package gg.climb.lolobjects.esports;

import gg.climb.lolobjects.RiotId;

import org.immutables.value.Value;

/**
 * A league format such as LCK or NALCS. Create an instance via
 * {@link ImmutableLeague#of(RiotId, String)}
 */
@Value.Immutable(builder = false)
public abstract class League {
  @Value.Parameter public abstract RiotId getId();

  @Value.Parameter public abstract String getName();
}

File: app/gg/climb/fifthdrake/lolobjects/game/League.scala
package gg.climb.fifthdrake.lolobjects.game

import gg.climb.fifthdrake.lolobjects.{InternalId, RiotId}

class League(val id: InternalId[League], val name: String, val riotId: RiotId[League]){

  override def toString: String = s"id=$id, name=$name, riotId=$riotId)"
}
